src/com/epam/javatrainings/array/ArrayUtil.java

```diff
@@ -10,23 +12,89 @@
      *
      */
     public void sortByOddAndEven(int[] array) {
+        // Space - O(1)
+        // Time - O(n^2)
+
+        // I am assuming that is not required that the result array should start with odd number,
+        // otherwise we can comment out the following block, which will not affect on comlexity
+        // if (array.length > 0 && array[0] % 2 == 0) {
+        //     for (int i = 1; i < array.length; i++) {
+        //         if (array[i] % 2 == 1) {
+        //             swap(array, i, 0);
+        //             break;
+        //        }
+        //    }
+        // }
+
+        for (int i = 0; i < array.length; i++) {
```

Comment:
check ones on each iteration is the "i" reach the array last index
for (int i = 0; i < array.length - 1; i++) {